### PR TITLE
Cleanup Divider Component

### PR DESCRIPTION
**Pull request by tiffwong**
## Description

This PR moves the basic styles to the component itself and also added props to easily adjust the spacing above and below the divider without having to create classes for the styles every time.

### CODE REVIEW

File: packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/LinodeNetworkingIPTransferPanel.tsx [hunk 1]
```diff
@@ -1,660 +0,0 @@
-import { assignAddresses } from '@linode/api-v4/lib/networking';
```
Comment (tiffwong) [inline packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/LinodeNetworkingIPTransferPanel.tsx:1]: Realized this file wasn't being used anywhere so I deleted it

File: packages/manager/src/components/core/Divider.tsx [hunk 2]
```diff
@@ -5,28 +5,29 @@ import * as React from 'react';
...
   },
 }));
 
 /* eslint-disable-next-line */
 export interface DividerProps extends _DividerProps {}
 
 interface Props extends _DividerProps {
-  type?: 'landingHeader' | 'other';
+  spacingTop?: number;
+  spacingBottom?: number;
```
Comment (HanaXu) [inline packages/manager/src/components/core/Divider.tsx:20]: Can we rename spacing to margin?

**Review by HanaXu — COMMENTED** [review]
Left some comments on dark mode!

File: packages/manager/src/components/core/Divider.tsx (hunk 2 above)
Comment (tiffwong) [reply packages/manager/src/components/core/Divider.tsx:25]: We use `spacingTop` as a custom prop for a couple other components so I'm going to leave it as is so it's consistent

**Review by DevDW — APPROVED** [review]
Went through the app and everything I observed looked good 👍

**Review by HanaXu — APPROVED** [review]
LGTM! 👍

**Pull request merged**